src/tree/mod.rs

```diff
@@ -49,7 +49,7 @@ mod test_treemath;
 #[cfg(test)]
 mod test_util;
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```

Comment:
Yes, we have to make a complete copy of the tree, including the of the secrets. There might be a way around it, but it would require a crazy setup of references, and at this point I want to keep it simple.

Reply:
Any change like this will cost us a couple weeks to fix down the road. If you think that's worth it, you can keep it as is.

Reply:
We can discuss this offline, but I don't see a big issue with copying the tree right now. This is also what the spec is suggesting.